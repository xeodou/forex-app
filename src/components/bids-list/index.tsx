import { CurrencyRate } from "lib/forex-api/client";
import { FC } from "react";

export const BidsList: FC<{ rates: CurrencyRate[] }> = ({ rates }) => {
  return (
    <div className="bids-list w-full">
      <header className="flex justify-around uppercase">
        <span>CUR</span>
        <span>Time</span>
        <span>Ask</span>
        <span>Bid</span>
      </header>
      <div className="h-60 overflow-y-auto flex flex-col">
        {rates.map((rate, index) => (
          <ul
            key={rate.time_stamp}
            className={`flex justify-around ${index % 2 === 0 ? "bg-slate-300" : "bg-slate-100 "}`}
          >
            <li>{`${rate.from}/${rate.to}`}</li>
            <li>{new Date(rate.time_stamp).toLocaleString()}</li>
            <li>{rate.ask}</li>
            <li>{rate.bid}</li>
          </ul>
        ))}
      </div>
    </div>
  );
};
