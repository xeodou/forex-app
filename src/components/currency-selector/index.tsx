import { FC } from "react";
import { currencies } from "utils/currencies";

export interface CurrencyHeaderProps {
  // On selected currency changes
  onSelected?: (currency: string) => void;
  // Specific the default selected currency
  value?: string;
}

export const CurrencySelector: FC<CurrencyHeaderProps> = ({ onSelected, value }) => {
  const onChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
    onSelected && onSelected(event.target.value);
  };

  return (
    <select
      onChange={onChange}
      value={value}
      className="h-full py-0 pl-4 pr-7 border-transparent bg-transparent text-gray-500 sm:text-sm "
    >
      {currencies.map((currency) => (
        <option key={currency} value={currency}>
          {currency}
        </option>
      ))}
    </select>
  );
};
