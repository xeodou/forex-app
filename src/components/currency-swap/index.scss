.currency-header {
  &--indicator {
    width: 30px;
  }
}
