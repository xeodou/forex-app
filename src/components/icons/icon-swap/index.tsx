export const IconSwap = () => {
  return (
    <svg
      xmlns="http://www.w3.org/2000/svg"
      className="w-6 h-6"
      fill="none"
      viewBox="0 0 24 24"
      stroke="currentColor"
      strokeWidth="1"
    >
      <path strokeLinecap="round" strokeLinejoin="round" d="M8 7h12m0 0l-4-4m4 4l-4 4m0 6H4m0 0l4 4m-4-4l4-4" />
    </svg>
  );
};
