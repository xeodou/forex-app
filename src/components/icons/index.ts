export * from "./icon-swap";
export * from "./icon-refresh";
export * from './icon-close';
