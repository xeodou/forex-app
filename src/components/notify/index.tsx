import { useNotify } from "hooks";
import { Item } from "./_item";

export const Notify = () => {
  const { stack, removeNotifyItem } = useNotify();

  const lastItem = stack[stack.length - 1];

  const onDismiss = (id: number) => {
    removeNotifyItem(id);
  };

  return <div className="notify-container">{lastItem && <Item item={lastItem} onDismiss={onDismiss} />}</div>;
};
