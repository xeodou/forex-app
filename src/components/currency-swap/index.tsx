import { CurrencySelector } from "components/currency-selector";
import { IconSwap } from "components/icons";
import { useCurrencyPair } from "hooks/use-currency-pair";

export const CurrencySwap = ({}) => {
  const [currencyPair, setCurrencyPair] = useCurrencyPair();

  const onFromCurrencyChange = (from: string) => {
    setCurrencyPair([from, currencyPair[1]]);
  };

  const onToCurrencyChange = (to: string) => {
    setCurrencyPair([currencyPair[0], to]);
  };

  const onSwitchClick = () => {
    setCurrencyPair([currencyPair[1], currencyPair[0]]);
  };

  return (
    <div className="flex items-center">
      <CurrencySelector onSelected={onFromCurrencyChange} value={currencyPair[0]} />
      <button
        type="button"
        aria-label="Switch currencies"
        className="font-medium text-sm p-2.5 text-center inline-flex items-center"
        onClick={onSwitchClick}
      >
        <IconSwap></IconSwap>
      </button>
      <CurrencySelector onSelected={onToCurrencyChange} value={currencyPair[1]} />
    </div>
  );
};
