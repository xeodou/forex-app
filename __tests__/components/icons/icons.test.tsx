import { act } from "@testing-library/react-hooks";
import { IconFresh, IconSwap } from "components/icons";
import { render, unmountComponentAtNode } from "react-dom";
import pretty from "pretty";

describe("icons", () => {
  let container: HTMLDivElement;

  beforeEach(() => {
    container = document.createElement("div");
    document.body.appendChild(container);
  });

  afterEach(() => {
    unmountComponentAtNode(container);
    container.remove();
    container = null as unknown as HTMLDivElement;
  });

  describe("icon-swap", () => {
    it("should render the icon", () => {
      act(() => {
        render(<IconSwap />, container);
      });

      expect(pretty(container.innerHTML)).toMatchSnapshot();

      expect(container.querySelector("svg")).toBeTruthy();
    });
  });

  describe("icon-fresh", () => {
    it("should render the icon", () => {
      act(() => {
        render(<IconFresh />, container);
      });

      expect(pretty(container.innerHTML)).toMatchSnapshot();

      expect(container.querySelector("svg")).toBeTruthy();
    });
  });
});
