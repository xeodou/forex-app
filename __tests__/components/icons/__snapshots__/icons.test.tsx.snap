// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`icons icon-close should render the icon 1`] = `
"<svg xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"h-6 w-6\\" fill=\\"none\\" viewBox=\\"0 0 24 24\\" stroke=\\"currentColor\\" stroke-width=\\"2\\">
  <path stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" d=\\"M6 18L18 6M6 6l12 12\\"></path>
</svg>"
`;

exports[`icons icon-fresh should render the icon 1`] = `
"<svg xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"h-6 w-6\\" fill=\\"none\\" viewBox=\\"0 0 24 24\\" stroke=\\"currentColor\\" stroke-width=\\"1\\">
  <path stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" d=\\"M4 4v5h.582m15.356 2A8.001 8.001 0 004.582 9m0 0H9m11 11v-5h-.581m0 0a8.003 8.003 0 01-15.357-2m15.357 2H15\\"></path>
</svg>"
`;

exports[`icons icon-swap should render the icon 1`] = `
"<svg xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"w-6 h-6\\" fill=\\"none\\" viewBox=\\"0 0 24 24\\" stroke=\\"currentColor\\" stroke-width=\\"1\\">
  <path stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" d=\\"M8 7h12m0 0l-4-4m4 4l-4 4m0 6H4m0 0l4 4m-4-4l4-4\\"></path>
</svg>"
`;
